README.md

```diff
@@ -76,15 +76,13 @@ You need to accept the model license before downloading or using the Stable Diff
 
 ```python
 # make sure you're logged in with `huggingface-cli login`
-from torch import autocast
 from diffusers import StableDiffusionPipeline
 
 pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4", use_auth_token=True)
 pipe = pipe.to("cuda")
 
 prompt = "a photo of an astronaut riding a horse on mars"
-with autocast("cuda"):
-    image = pipe(prompt).images[0]  
+image = pipe(prompt).images[0]  
```

Comment:
Are you sure that this is faster? Using `autocast` gives currently (before this PR) a 2x boost in terms of generation speed. 

Will also test a bit locally on a GPU tomorrow

Reply:
This is extremely surprising but I am also measuring a 2x speedup with autocast on `f32`.

I am looking into it, I see to copies but not nearly the same amount, there's probably a device-to-host /host-to-device somewhere that kills performance but I haven't found it yet.

Reply:
Okay, I figure it out.

`autocast` will actually use `fp16` for some ops by doing some heuristics. https://pytorch.org/docs/stable/amp.html#cuda-op-specific-behavior

So it's faster because it's running on fp16 even if the model was loaded in f32.

If we enable *real* fp16 with a big performance boost I feel like we shouldn't need it f32 (but that does make it slower but also "more" correct.)

But it's still a nice way to get f16 running "for free". The heuristics they use seem OK (but as they mention, they probably wouldn't work for gradients, I guess because fp16 overflows faster)

Reply:
Commented extensively here: https://github.com/huggingface/diffusers/pull/511/#issuecomment-1268352935

Should we maybe change the example to lead native `fp16` weights?

Reply:
Yes I think fp16 is widely supported now